presto-main/src/test/java/com/facebook/presto/type/BenchmarkBigIntOperators.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongUnaryOperator;
+
+@State(Scope.Thread)
+@Fork(2)
+@Warmup(iterations = 100, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 100, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.Throughput)
+public class BenchmarkBigIntOperators
+{
+    private long[] leftOperands;
+    private long[] rightOperands;
+
+    @Setup
+    public void setup()
+    {
+        leftOperands = new long[] { 1, 20, 33, 407, 7890 };
+        rightOperands = new long[] { 123456, 9003, 809, 67, 5};
+    }
+
+    @Benchmark
+    public Object overflowChecksAdd()
+    {
+        return execute((x, y) -> BigintOperators.add(x, y));
```

Comment:
@martint pointed out to me that the lambda is not guaranteed to be optimized away, so I would rewrite all of these to not use a lambda. Probably good to change `leftOperands` and `rightOperands` to `long` rather than `long[]` too, so that we're not measuring the array access time at all